flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/sink/DynamoDbSinkWriter.java

```diff
@@ -149,7 +150,7 @@ public DynamoDbSinkWriter(
                 maxRecordSizeInBytes,
                 states);
         this.failOnError = failOnError;
-        this.tableName = tableName;
+        this.table = tableArn != null ? tableArn : tableName;
```

Comment:
What if both parameters are set? If we rely on the Builder class to run the validations, should we make this method package private? 

Nit: can we make the name more explicit? Something like tableArnOrName?